sympy/simplify/cse_main.py

```diff
@@ -40,6 +40,195 @@
+    {x0: x1/x2, x3: x0 - exp(x2)}
+
+    Parameters
+    ==========
+
+    replacements : ``List[Tuple[Symbol, Expr]]``
+        The symbol-expression replacements pairs resulting from the CSE.
+    reduced_exprs : ``List[Expr]``
+        The replaced expressions resulting from the CSE.
+    replacements_mapping : ``Dict[Symbol, Expr]``
+        A mapping of replacements symbols to their corresponding expression.
+
+
+    See Also
+    ========
+
+    cse: Function for common subexpression elimination.
+
+    """
+
+    @classmethod
+    def from_reduced_exprs(cls, reduced_exprs, replacements_mapping, *,
+                           as_list=True):
+        """Alternate constructor to instantiate a ``CseExpr`` from a list of
+        reduced expressions and a mapping of replacements.
+
+        Examples
+        ========
+
+        A ``CseExpr`` can be constructed from a reduced expression and a
+        dictionary of replacements mapping SymPy symbols to SymPy expressions:
+
+        >>> from sympy import symbols, sin, exp
+        >>> from sympy.simplify.cse_main import CseExpr
+        >>> x0, x1, x2, x3 = symbols('x0, x1, x2, x3')
+        >>> reduced_expr = x3 * (x3 + sin(x0))
+        >>> replacements_mapping = {x0: x1 / x2, x3: x0 - exp(x2)}
+        >>> cse_expr = CseExpr.from_reduced_exprs(reduced_expr,
+        ...                                       replacements_mapping)
+        >>> print(cse_expr)
+        ([(x0, x1/x2), (x3, x0 - exp(x2))], [x3*(x3 + sin(x0))])
+
+        As only a single expression is passed as an argument to
+        ``reduced_exprs``, this method will wrap it into a single-item ``list``.
+        Alternatively, the reduced expression can be kept unchanged using the
+        ``as_list`` parameter:
+
+        >>> cse_expr = CseExpr.from_reduced_exprs(reduced_expr,
+        ...                                       replacements_mapping,
+        ...                                       as_list=False)
+        >>> print(cse_expr)
+        ([(x0, x1/x2), (x3, x0 - exp(x2))], x3*(x3 + sin(x0)))
+
+        Parameters
+        ==========
+
+        reduced_exprs : ``Expr`` or ``List[Expr]``
+            The reduced expression(s).
+        replacements_mapping : ``Dict[Symbol, Expr]``
+            Pairs of replacements present in ``exprs``.
+        as_list : ``bool``, default is ``True``
+            Whether the argument passed to ``reduced_exprs`` should be converted
+            to a ``list``. With the default of ``True``, if an iterable is
+            passed to ``reduced_exprs`` then it is converted to a ``list``, or
+            if a single ``Expr`` is passed then it is wrapped into a single-item
+            ``list``. If ``False``, then the argument passed to
+            ``reduced_exprs`` is left as is, meaning that the value returned by
+            the ``reduced_exprs`` property is not necessarily a ``list``. This
+            parameter is similar to the ``list`` parameter of the ``cse``
+            function.
```

Comment:
This parameter is messy. It should be for the caller to pass a list or not or there should be a `CseExprList` type or something to represent the list case.

The simplest solution is to say that the caller should always pass a list.